Modules/PostFilter/ShutterFilter/Output/ShuttterFilterOutput.cs

```diff
@@ -0,0 +1,88 @@
+﻿using Vixen.Commands;
+using Vixen.Data.Evaluator;
+using Vixen.Data.Value;
+using Vixen.Intent;
+using Vixen.Sys;
+using Vixen.Sys.LayerMixing;
+using VixenModules.OutputFilter.TaggedFilter.Outputs;
+
+namespace VixenModules.OutputFilter.ShutterFilter.Output
+{
+	/// <summary>
+	/// Shutter filter output.  This output contains special conversion logic for fixtures.
+	/// This output extract the intensity from color intents and creates open shutter commands.
+	/// </summary>
+	public class ShutterFilterOutput : TaggedFilterOutputBase<Filter.ShutterFilter>
+	{
+		#region Fields
+		
+		/// <summary>
+		/// Open Shutter Index command value.
+		/// </summary>
+		private byte _openShutterIndexValue;
+
+		#endregion
+
+		#region Constructor
+
+		/// <summary>
+		/// Constructor
+		/// </summary>
+		/// <param name="tag">Tag (name) of shutter function</param>		
+		/// <param name="openShutterIndexValue">Index value for shutter open command</param>
+		public ShutterFilterOutput(
+			string tag, 			
+			byte openShutterIndexValue) : base(tag)			
+		{			
+			// Store off the open shutter index value
+			_openShutterIndexValue = openShutterIndexValue;
+		}
+
+		#endregion
+
+		#region Protected Methods
+
+		/// <summary>
+		/// Process the input intent.
+		/// </summary>
+		/// <param name="intent">Intent to process</param>
+		protected override void ProcessInputDataInternal(IIntentState intent)
+		{
+			// If the intent is a discrete color then...
+			if (intent is IIntentState<DiscreteValue> ||
+			    intent is IIntentState<RGBValue> ||
+			    intent is IIntentState<LightingValue>)
+			{
```

Comment:
Is there a reason that the dispatch / handle pattern was not used here?

Reply:
Hmm that is an interesting question.  All of the filters I created are based off the pan/tilt filter you created last year.  I think my take away from the pan/tilt filter was that the dispatch handle pattern was being used to determine if the incoming intent was applicable to the output.  The shutter filter is also doing this via the dispatch / handle pattern.  The unique thing about the shutter filter is that it has multiple intents that are applicable and they need to handled differently.  If someone manually created shutter tagged intents they can just forwarded along.   If color intents are coming in they need to be converted to shutter intents. 

TaggedFilterOutputBase.cs - ProcessInputData method shows the pattern I tried to capture from your work.

I could create a second dispatch / handle class that either passes the intent along or does the conversion to a shutter intent.  This dispatch / handle would need a reference to the output class IntentData member.

Alternatively I could deviate ProcessInputData method / pattern and just have one filter that determines if the intent is applicable and then goes and handles it directly.

Please respond with your preference and any nuggets of wisdom.

Reply:
Looks like I did the same thing in the dimming filter.

Reply:
I am thinking I should move the logic to convert the shutter to a command intent into the existing filter.

Reply:
I moved all the intent conversion logic into the shutter filter class.  The dispatch / handle pattern is being fully leveraged now.

Reply:
I added the close shutter value.  This material is ready for further review.